src/sql/src/statements.rs

```diff
@@ -58,6 +59,26 @@ use crate::error::{
     SerializeColumnDefaultConstraintSnafu, TimestampOverflowSnafu, UnsupportedDefaultValueSnafu,
 };
 
+const REDACTED_OPTIONS: [&str; 2] = ["access_key_id", "secret_access_key"];
+
+fn redact_and_sort_options(options: &OptionMap) -> Vec<String> {
+    let mut result = vec![];
+    let keys = options.as_ref().keys().sorted();
+    for key in keys {
+        if let Some(val) = options.get(key) {
+            let redacted = REDACTED_OPTIONS
+                .iter()
+                .any(|opt| opt.eq_ignore_ascii_case(key));
+            if redacted {
+                result.push(format!("{key} = ******"));
+            } else {
+                result.push(format!("{key} = {val}"));
```

Comment:
Looks better to add a quote for values:
```suggestion
                result.push(format!("{key} = '{val}'"));
```